pkg/kubectl/cmd/scale.go

```diff
@@ -153,6 +155,21 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 			errs = append(errs, err)
 			continue
 		}
+		if cmdutil.GetRecordFlag(cmd) {
```

Comment:
For all commands: We should add/update the annotation if the flag is set or if the annotation is already on the object.

Something like we do for apply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/apply.go#L192